modules/tinymce/src/core/main/ts/dom/TrimNode.ts

```diff
@@ -5,20 +5,30 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Arr, Strings } from '@ephox/katamari';
 import { SugarElement } from '@ephox/sugar';
-import Tools from '../api/util/Tools';
-import * as ElementType from './ElementType';
+import DOMUtils from '../api/dom/DOMUtils';
 import * as NodeType from './NodeType';
+import * as Empty from './Empty';
 
-const surroundedBySpans = function (node) {
-  const previousIsSpan = node.previousSibling && node.previousSibling.nodeName === 'SPAN';
-  const nextIsSpan = node.nextSibling && node.nextSibling.nodeName === 'SPAN';
+const isSpan = (node: Node): node is HTMLSpanElement =>
+  node && node.nodeName.toLowerCase() === 'span';
```

Comment:
Are we expecting people to pass null or undefined in here? I think we should either remove the `node &&` or be explicit with the types that we're expecting `node?: Node` or `node: Node | null`

Reply:
Update, given we want to use it with previousSibling, it should be `node: Node | null`... not sure how it's compiling now, that's a worry

Reply:
Take a look at the latest changes... It's no longer used for that, however you are right we don't need the `node &&` check anymore.

Reply:
I saw the new commit just as I was submitting the review. Latest version looks fine.